src/main/java/frc/robot/command/AutoAlignCommand.java

```diff
@@ -0,0 +1,217 @@
+package frc.robot.command;
+
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.trajectory.Trajectory;
+import edu.wpi.first.math.trajectory.TrajectoryConfig;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.ParallelRaceGroup;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import edu.wpi.first.wpilibj2.command.WaitUntilCommand;
+import frc.robot.Constants;
+import frc.robot.input.ControllerInput;
+import frc.robot.subsystem.DriveSubsystem;
+import frc.robot.subsystem.PositioningSubsystem;
+import frc.robot.utils.Vec2;
+import com.pathplanner.lib.PathPlanner;
+
+/*
+ Driver holds down one of three buttons
+ The robot finds the closest master node
+ The robot generates a rotational and translational absolute trajectory to closest master node
+ The robot generates a translation relavative trajectory to the sub node based button held by the driver
+ The command ends when the final position is reached.
+ */
+public class AutoAlignCommand extends Command {
+  private final DriveSubsystem drive;
+  private final PositioningSubsystem positioning;
+  private final ControllerInput controller;
+
+  // Selected auto-align mode from the driver's button press
+  private AutoAlignMode selectedMode;
+
+  // Trajectories for the two phases
+  private Trajectory absoluteTrajectory;
+  private Trajectory relativeTrajectory;
+
+  // A sequential command that will follow both trajectories
+  private Command trajectoryCommand;
+
+  // Example master node positions (12 master nodes for 6 reef sides, 2 per side).
+  // These should normally be read from an external configuration (e.g. Elastic Dashboard)
+  // Here we provide dummy sample values (in meters) in field coordinates.
+  private final Pose2d[] masterNodes = new Pose2d[] {
+      new Pose2d(2.0, 7.0, Constants.ZERO_ROTATION),
+      new Pose2d(4.0, 7.0, Constants.ZERO_ROTATION),
+      new Pose2d(6.0, 7.0, Constants.ZERO_ROTATION),
+      new Pose2d(2.0, 5.0, Constants.ZERO_ROTATION),
+      new Pose2d(4.0, 5.0, Constants.ZERO_ROTATION),
+      new Pose2d(6.0, 5.0, Constants.ZERO_ROTATION),
+      new Pose2d(2.0, 3.0, Constants.ZERO_ROTATION),
+      new Pose2d(4.0, 3.0, Constants.ZERO_ROTATION),
+      new Pose2d(6.0, 3.0, Constants.ZERO_ROTATION),
+      new Pose2d(2.0, 1.0, Constants.ZERO_ROTATION),
+      new Pose2d(4.0, 1.0, Constants.ZERO_ROTATION),
+      new Pose2d(6.0, 1.0, Constants.ZERO_ROTATION)
+  };
```

Comment:
Not sure these need to be read from a dashboard. They shouldn't ever change since they're inherent to this year's field layout. So you should be able to hard-code them here.

Reply:
Also, `Constants.ZERO_ROTATION` is a hallucination and doesn't exist